### PR TITLE
[BLB] Daring Waverider, Hoarder's Overflow, Driftgloom Coyote, Lilysplash Mentor

### PR DESCRIPTION
**Pull request by ssk97**
Feels like Lilysplash Mentor should be made into a variant of `ExileThenReturnTargetEffect`, but this works for now I think.

Edit: Can't actually do it correctly with the changes from #12605, since the counter needs to be applied before the creature is back on the battlefield.

### CODE REVIEW

File: Mage.Sets/src/mage/cards/d/DaringWaverider.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    public DaringWaverider(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{4}{U}{U}");
+
+        this.subtype.add(SubType.OTTER);
+        this.subtype.add(SubType.WIZARD);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // When Daring Waverider enters, you may cast target instant or sorcery card with mana value 4 or less from your graveyard without paying its mana cost. If that spell would be put into your graveyard, exile it instead.
+        Ability ability = new EntersBattlefieldTriggeredAbility(new MayCastTargetCardEffect(CastManaAdjustment.WITHOUT_PAYING_MANA_COST, true));
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/d/DaringWaverider.java:39]: I think this is going to result in double "you may" and two choices to click through

**Review by xenohedron — APPROVED** [review]
I have some ideas for moving lilysplash mentor to common class but this is fine as is

File: Mage.Sets/src/mage/cards/d/DaringWaverider.java (hunk 1 above)
Comment (ssk97) [reply Mage.Sets/src/mage/cards/d/DaringWaverider.java:39]: `EntersBattlefieldTriggeredAbility` optional is false, the `true` there is for `MayCastTargetCardEffect`'s `thenExile`

Comment (xenohedron) [reply Mage.Sets/src/mage/cards/d/DaringWaverider.java:39]: my bad, sorry

**Pull request merged**